server/src/main/java/io/crate/expression/reference/file/LineContext.java

```diff
@@ -53,11 +55,13 @@ String sourceAsString() {
     }
 
     @Nullable
-    Map<String, Object> sourceAsMap() {
+    LinkedHashMap<String, Object> sourceAsMap() {
         if (parsedSource == null) {
             if (rawSource != null) {
                 try {
-                    parsedSource = XContentHelper.toMap(new BytesArray(rawSource), XContentType.JSON);
+                    // preserve the order of the rawSource
+                    ParsedXContent parsedXContent = XContentHelper.convertToMap(new BytesArray(rawSource), true, XContentType.JSON);
+                    parsedSource = new LinkedHashMap<>(parsedXContent.map());
```

Comment:
`parsedXContent.map()` should already return a `LinkedHashMap` since `convertToMap()` is called with `true`, so why not just cast and avoid the instantiation?

Reply:
Thanks 👍 will make the change